canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx

```diff
@@ -75,6 +75,8 @@ class ButtonAction extends React.Component {
 		const disabled = this.props.state === STATES.DISABLED;
 		const actionButtonKind = this.getActionButtonKind();
 		const actionButtonSize = this.getActionButtonSize();
+		// Get the icon to be rendered from the call back
+		const icon = this.props.buttonIconHandler?.(this.props.action.name, this.props.action);
```

Comment:
This doesn't need to be a new prop.  You can get the handler from the controller.